docs/user_guide/backend_configuration.rst

```diff
@@ -0,0 +1,286 @@
+    from uuid import uuid4
+
+    from dateutil import tz
+    from neuroconv.tools.nwb_helpers import get_default_backend_configuration
+    from pynwb import NWBFile, TimeSeries
+
+    session_start_time = datetime(2020, 1, 1, 12, 30, 0, tzinfo=tz.gettz("US/Pacific"))
+    nwbfile = NWBFile(
+        identifier=str(uuid.uuid4()),
+        session_start_time=session_start_time,
+        session_description="A session of my experiment.",
+    )
+
+    time_series = TimeSeries(
+        name="MyTimeSeries",
+        description="A time series from my experiment.",
+        unit="cm/s",
+        data=[1., 2., 3.],
+        timestamps=[0.0, 0.2, 0.4],
+    )
+    nwbfile.add_acquisition(time_series)
+
+    default_backend_configuration = get_default_backend_configuration(
+        nwbfile=nwbfile, backend="hdf5"
+    )
+
+From which a printout of the contents looks like...
+
+.. code-block:: python
+
+    print(default_backend_configuration)
+
+.. code-block:: bash
+
+    Configurable datasets identified using the hdf5 backend
+    -------------------------------------------------------
+
+    acquisition/MyTimeSeries/data
+    -----------------------------
+      dtype : float64
+      full shape of source array : (3,)
+      full size of source array : 0.00 GB
+
+      buffer shape : (3,)
+      expected RAM usage : 0.00 GB
+
+      chunk shape : (3,)
+      disk space usage per chunk : 0.00 MB
+
+      compression method : gzip
+
+
+    acquisition/MyTimeSeries/timestamps
+    -----------------------------------
+      dtype : float64
+      full shape of source array : (3,)
+      full size of source array : 0.00 GB
+
+      buffer shape : (3,)
+      expected RAM usage : 0.00 GB
+
+      chunk shape : (3,)
+      disk space usage per chunk : 0.00 MB
+
+      compression method : gzip
+
+
+Customization
+-------------
+
+To modify the chunking or buffering patterns and compression method or options, change those values in the ``.dataset_configurations`` object using the location of each dataset as a specifier.
+
+Let's demonstrate this by modifying everything we can for the ``data`` field of the ``TimeSeries`` object generated above...
+
+.. code-block:: python
+
+    dataset_configurations = default_backend_configuration.dataset_configurations
+    dataset_configuration = dataset_configurations["acquisition/MyTimeSeries/data"]
+    dataset_configuration.chunk_shape = (1,)
+    dataset_configuration.buffer_shape = (2,)
+    dataset_configuration.compression_method = "Zstd"
+    dataset_configuration.compression_options = dict(clevel=3)
+
+Some details to note about what can be changed...
+
+.. note::
+
+    Core fields such as the maximum shape and data type of the source data cannot be altered using this method.
+
+.. note::
+
+    The ``buffer_shape`` must be a multiple of the ``chunk_shape`` along each axis.
+
+.. note::
+
+    You can see what compression methods are available on your installation by examining the following...
```

Comment:
What about the options for each method? Should we add a comment on how to fetch or figure out the options for each method?

Reply:
At this point there is no strategy for automatic fetching or even 'figuring out' easily, so I wouldn't even mention it

Eventually maybe we can put together a few hand written schemas for these

Reply:
We're already linking to the closest possible page at the top; otherwise that info is scattered on various dynamic filter pages per method

Reply:
I see. OK, then for def out of scope.